common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signup/SignUpStartApiResponse.kt

```diff
@@ -0,0 +1,169 @@
+    @Expose @SerializedName("unverified_attributes") val unverifiedAttributes: List<Map<String, String>>?,
+    @Expose @SerializedName("invalid_attributes") val invalidAttributes: List<Map<String, String>>?,
+    @SerializedName("details") val details: List<Map<String, String>>?,
+    @Expose @SerializedName("challenge_type") val challengeType: String?,
+    @Expose @SerializedName("error_codes") val errorCodes: List<Int>?
+) : IApiResponse(statusCode) {
+
+    companion object {
+        private val TAG = SignUpStartApiResponse::class.java.simpleName
+    }
+
+    fun toResult(): SignUpStartApiResult {
+        LogSession.logMethodCall(TAG, "${TAG}.toResult")
+
+        return when (statusCode) {
+
+            // Handle 400 errors
+            HttpURLConnection.HTTP_BAD_REQUEST -> {
+                when {
+                    error.isUserAlreadyExists() -> {
+                        SignUpStartApiResult.UsernameAlreadyExists(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    errorCodes?.get(0).isInvalidParameter() and (errorDescription?.isInvalidEmail() == true) -> {
+                        SignUpStartApiResult.InvalidEmail(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    error.isAuthNotSupported() -> {
+                        SignUpStartApiResult.AuthNotSupported(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    error.isAttributeValidationFailed() -> {
+                        SignUpStartApiResult.InvalidAttributes(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty(),
+                            invalidAttributes = invalidAttributes?.toAttributeList()
+                                ?: return SignUpStartApiResult.UnknownError(
+                                    error = "invalid_state",
+                                    errorDescription = "SignUp /start did not return a invalid_attributes with validation_failed error",
+                                    details = details
+                                )
+                        )
+                    }
+                    error.isUnsupportedChallengeType() -> {
+                        SignUpStartApiResult.UnsupportedChallengeType(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    error.isPasswordTooWeak() || error.isPasswordTooLong() || error.isPasswordTooShort()
```

Comment:
Ok, so the password strength verification is happening on the server?

Reply:
yes